packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts

```diff
@@ -1239,6 +1247,9 @@ export class Service extends cdk.Resource implements iam.IGrantable {
           this.renderCodeConfiguration(this.source.codeRepository!.codeConfiguration.configurationValues!) :
           undefined,
       },
+      encryptionConfiguration: this.props.kmsKey ? {
+        kmsKey: this.props.kmsKey.keyArn,
+      } : undefined,
```

Comment:
I think we need to update the role assigned to the App runner to use the KMS key. I will run some test to verify this.

Reply:
Thank you for the review!

I also thought the same, but when I actually tried deploying using a CMK, it was able to host without any issues, so I had concluded that no additional permissions were required.

I may have made a mistake somewhere, so I will double check again. If you happen to find out anything else, please let me know as well.

Reply:
did some testing, and it looks like during the App Runner service creation, CreateGrant method got created on the passed KMS that should allow this action for the account Principal or the role that create the CFN stack. so, I think this grant is not required.